src/types/project.types.ts

```diff
@@ -48,16 +48,23 @@ export interface ProjectContributorIdentity {
 
 export type ProjectContributorType = 'ProjectManager' | 'ProjectParticipant';
 
-export interface ProjectContributor {
+export interface ProjectContributorRole {
   type: ProjectContributorType;
   affiliation: ProjectOrganization;
+}
+export interface ProjectContributor {
   identity: ProjectContributorIdentity;
+  roles: ProjectContributorRole[];
 }
 
-interface ProjectCreator extends Omit<ProjectContributor, 'type'> {
+export interface CreatorRole extends Omit<ProjectContributorRole, 'type'> {
   type: 'ProjectCreator';
 }
 
+interface ProjectCreator extends Omit<ProjectContributor, 'roles'> {
+  roles: [CreatorRole];
```

Comment:
Tenkte først at dette var en feil, men det er kanskje en notasjon for å vise at det første (eneste?) elementet er av typen `CreatorRole`? Jeg tenkte først at det skulle vært `CreatorRole[]`, men det vil være mer upresist i så fall

Reply:
Jeg finner ikke igjen dokumentasjon på det, men mener at din tolkning (og min implementasjon) er riktig, altså at [CreatorRole] betyr at det er ett element av typen CreatorRole. Fant ikke igjen ts-dokumentasjon på det, men spurte KI-chat som også tolket det sånn. Jeg er usikker på om det er rett her, altså om creator kan ha flere roller. Men jeg har ikke så mange eksempler å se til. Kan dobbeltsjekke med TA og endre om nødvendig.

Reply:
Ta bekreftet at creator alltid vil ha rollen ProjectCreator